### PR TITLE
add additional headers option; support using javelin cell

### PR DESCRIPTION
**Pull request by rwillig**
Hi Matt, 

I wonder if there would be any support for this change in an official release of castra. We found a need for it at Adzerk/Kevel and forked and submoduled the version of castra with our changes, but obviously we'd prefer to continue pulling the dependency from clojars. Thanks in advance.

### CODE REVIEW

File: src/castra/core.cljs [hunk 1]
```diff
@@ -80,12 +80,15 @@
...
 
 (defn- ajax
-  [{:keys [ajax-fn clj->json json->clj on-error] :as opts} expr]
+  [{:keys [ajax-fn clj->json json->clj on-error additional-headers] :as opts} expr]
   (let [prom    (.Deferred js/jQuery)
         headers (-> {"X-Castra-Csrf"          "true"
                      "X-Castra-Tunnel"        "transit"
                      "X-Castra-Validate-Only" (str (boolean *validate-only*))
                      "Accept"                 "application/transit+json"}
+                    (merge (if (instance? javeline.core.Cell additional-headers)
```
Comment (daveyarwood) [inline src/castra/core.cljs:89]:
```suggestion
                    (merge (if (instance? javelin.core.Cell additional-headers)
```

@rwillig I found a typo :)

**Review by burn2delete — APPROVED** [review]

**Comment by flyboarder** [comment]
Thanks! I'll get a release pushed ASAP.

File: src/castra/core.cljs (hunk 1 above)
Comment (rwillig) [reply src/castra/core.cljs:89]: holy crap, how did that get past me?

Comment (rwillig) [reply src/castra/core.cljs:89]: that's actually weird bc, it must  be broken in our submodule, yet it works?

Comment (rwillig) [reply src/castra/core.cljs:89]: oh nvm. i had just copied the file over from our submodule but i probably didn't have our latest branch checked out

**Comment by daveyarwood** [comment]
@flyboarder Heads up about the typo I found! See above.

**Comment by burn2delete** [comment]
typo fixed!

**Pull request merged**